simple_ado/utilities.py

```diff
@@ -35,17 +34,21 @@ def download_from_response_stream(
     if response.status_code < 200 or response.status_code >= 300:
         raise ADOHTTPException("Failed to fetch file", response)
 
-    with open(output_path, "wb") as output_file:
-
-        content_length_string = response.headers.get("content-length", "0")
-
-        total_size = int(content_length_string)
-        total_downloaded = 0
-
-        for data in response.iter_content(chunk_size=chunk_size):
-            total_downloaded += len(data)
-            output_file.write(data)
-
-            if total_size != 0:
-                progress = int((total_downloaded * 100.0) / total_size)
-                log.info(f"Download progress: {progress}%")
+    if callback:
+        if callable(callback):
+            callback(chunk_size=chunk_size,response=response,output=output_path)
```

Comment:
The callback shouldn't handle the download. This method should still do that, it should just tell the callback about the process. Something like `callback(total_downloaded, total_size)`